modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/registry/RunningQueryInfoImpl.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.sql.engine.registry;
+
+import java.time.Instant;
+import java.util.Objects;
+import java.util.UUID;
+import org.apache.ignite.internal.sql.engine.AsyncSqlCursor;
+import org.apache.ignite.internal.sql.engine.SqlQueryType;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Information about a running query.
+ */
+public class RunningQueryInfoImpl implements RunningQueryInfo {
```

Comment:
What about working in a multithreaded environment? Do we have right now a guarantee that all set/get methods will be invoked from the single thread?

Reply:
All mutable fields are marked as `volatile` :thinking: